### PR TITLE
Add support for script files and refactor unpacker

### PR DESCRIPTION
**Pull request by nnguyen259**
Script files unpacker will unpack/split the original script files into chunk with header as described in the file's header.

Refactor tbl unpacker to support a more generic format.

### CODE REVIEW

**Review by hwjsnc — DISMISSED** [review]
Looks good to me!

**Comment by nnguyen259** [comment]
So my biggest concern is that the script packer/unpacker is really slow due to the amount of files. Wonder if there's a way to specify which file to pack/unpack instead of doing bulk like this.

**Comment by nnguyen259** [comment]
Also this will change the layout of the unpacked folder so existing projects will need to be regenerated.

**Comment by hwjsnc** [comment]
Yeah, I did notice it being slower. Depending on what's needed, it would be possible to pass `scriptTypes` as an argument to `unpack`, or perhaps `scriptPath`/`outPath` and `file`. I have no idea how to determine which files are interesting though, so I think the only way to go is probably to extract all of them anyway.

The changed paths aren't a problem for me, I can easily fix that downstream.

**Comment by nnguyen259** [comment]
gonna work on this some more with a new approach since doing bulk script pack/unpack absolutely kill the performance. New approach should enable the user to choose which module to unpack during the unpacking process and which specific file(s) to pack in the packing process. Randomizer will have a list of script files to pack as well for them and some module will be disabled if the specific script files aren't found/unpacked.

**Comment by nnguyen259** [comment]
ok when packing/unpacking you now have the option to ignore scripts, and also you can pick the specific script folder to pack/unpack. That would help the performance a bit. Next up is adding [Twn's SenScriptDecomplier](https://github.com/TwnKey/SenScriptsDecompiler) as an option

File: frames/packer.py [hunk 1]
```diff
@@ -66,26 +67,72 @@ def __init__(self, master=None):
...
+        if not self.packScript.get():
+            self.frmScriptSelect.grid_remove()
+        else:
+            self.frmScriptSelect.grid()
+
+    def sortAndRefresh(self):
+        self.availbleList.sort()
+        self.selectedList.sort()
+        self.availbleChoices.set(self.availbleList)
+        self.selectedChoices.set(self.selectedList)
```
Comment (hwjsnc) [inline frames/packer.py]: Is "availble" here a typo or just shortened so the identifiers have the same length?

File: frames/unpacker.py [hunk 2]
```diff
@@ -28,29 +28,102 @@ def __init__(self, master=None):
...
+
+        self.btnMoveRight = ttk.Button(self.frmScriptSelect, text=' > ', command=self.moveRight)
+        self.btnMoveRightAll = ttk.Button(self.frmScriptSelect, text=' >>> ', command=self.moveRightAll)
+        self.btnMoveLeft = ttk.Button(self.frmScriptSelect, text=' < ', command=self.moveLeft)
+        self.btnMoveLeftAll = ttk.Button(self.frmScriptSelect, text=' <<< ', command=self.moveLeftAll)
+
+        self.btnMoveRightAll.grid(row=2, column=1, padx=5, pady=5)
+        self.btnMoveRight.grid(row=3, column=1, padx=5, pady=5)
+        self.btnMoveLeft.grid(row=4, column=1, padx=5, pady=5)
+        self.btnMoveLeftAll.grid(row=5, column=1, padx=5, pady=5)
```
Comment (hwjsnc) [inline frames/unpacker.py]: The order that the buttons here are created in doesn't match their position in the grid, so keyboard navigation is a bit strange. I think moving `self.btnMoveRightAll` one line up should fix it.

File: frames/packer.py (hunk 1 above)
Comment (nnguyen259) [reply frames/packer.py]: definitely a typo

File: frames/unpacker.py (hunk 2 above)
Comment (nnguyen259) [reply frames/unpacker.py]: @hwjsnc Can you check if it's better now?

**Comment by nnguyen259** [comment]
Gonna merge after checking over the gui. decompiler will have to wait for later, merging this now will give us updated unpacker to support t_item_en and the like

File: frames/unpacker.py (hunk 2 above)
Comment (hwjsnc) [reply frames/unpacker.py]: Yes, looks good

**Pull request merged**